ui/dgt/src/play.ts

```diff
@@ -94,38 +94,42 @@ export default function (token: string) {
       console['old' + name] = console[name];
       //Rewire function
       console[name] = function () {
-        var output = "";
-        for (let i = 0; i < arguments.length; i++) {
-          let arg = arguments[i];
-          output += "<span class=\"log-" + (typeof arg) + " log-" + name + "\">";
-          if (
-            typeof arg === "object" &&
-            typeof JSON === "object" &&
-            typeof JSON.stringify === "function"
-          ) {
-            output += JSON.stringify(arg);
+        //Return a promise so execution is not delayed by string manipulation
+        return new Promise(resolve => {
+          var output = "";
+          for (let i = 0; i < arguments.length; i++) {
+            let arg = arguments[i];
+            output += "<span class=\"log-" + (typeof arg) + " log-" + name + "\">";
+            if (
+              typeof arg === "object" &&
+              typeof JSON === "object" &&
+              typeof JSON.stringify === "function"
```

Comment:
you really don't need to test `typeof JSON === "object" && typeof JSON.stringify === "function"` all targeted browsers have it